azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -426,10 +427,11 @@ public int selectAndUpdateExecutionWithLocking(final int executorId, final boole
       if (hasLocked) {
         try {
           final List<Integer> execIds = transOperator.query(selectExecutionForUpdate,
-              new SelectFromExecutionFlows(), Status.PREPARING.getNumVal(), executorId);
+              new SelectFromExecutionFlows(), Status.READY.getNumVal(), executorId);
           if (CollectionUtils.isNotEmpty(execIds)) {
             execId = execIds.get(0);
-            transOperator.update(UPDATE_EXECUTION, executorId, System.currentTimeMillis(), execId);
+            transOperator.update(UPDATE_EXECUTION, executorId, System.currentTimeMillis(),
```

Comment:
Can you please make a change in selectAndUpdateExecutionWithLocking(final boolean batchEnabled,
      final int limit, final Status updatedStatus) method? It should not expect status anymore. It will be consistent to this change that you are making.

Reply:
Let's take it in separate PR because couple of test files also will be changed.

Reply:
Like discussed offline, I will take care of this change in separate PR.